src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/MethodInfos/RuntimeSyntheticMethodInfo.cs

```diff
@@ -80,7 +80,7 @@ public sealed override MethodInfo GetGenericMethodDefinition()
 
         public sealed override int GetHashCode()
         {
-            return _declaringType.GetHashCode();
+            return HashCode.Combine(_declaringType.GetHashCode(), ReflectedType.GetHashCode());
```

Comment:
`ReflectedType` is not used in `Equals` (but it's used in the non AOT version)
Add `_syntheticMethodId` instead?